latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,197 @@
+                        """\
+                        run apt-get update -y && \\
+                            apt-get install -y curl && \\
+                            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \\
+                            mkdir /root/.conda && \\
+                            bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \\
+                            rm -f Miniconda3-latest-Linux-x86_64.sh && \\
+                            conda init bash \
+                        """
+                    ).strip(),
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+            DockerCmdBlock(
+                comment="build and configure conda environment",
+                commands=[
+                    "copy environment.yml /root/environment.yml",
+                    f"run conda env create -f environment.yml --name {env_name}",
+                    f"""shell ["conda", "run", "-n", "{env_name}", "/bin/bash", "-c"]""",
+                    "run /opt/conda/bin/pip install --upgrade latch",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+        ]
+
+    # python package in workflow directory (poetry, distutils, etc.)
+    if (pkg_root / "setup.py").exists() or (pkg_root / "pyproject.toml").exists():
+        print("Add local package to python environment")
+        commands.append(
+            DockerCmdBlock(
+                comment="add local package to python environment",
+                commands=["run python3 -m pip install -e /root/"],
+                order=DockerCmdBlockOrder.POSTCOPY,
+            )
+        )
+
+    # python requirements.txt
+    if (pkg_root / "requirements.txt").exists():
+        print("Add requirements from `requirements.txt` to python environment")
+        commands.append(
+            DockerCmdBlock(
+                comment="add requirements from `requirements.txt` to python environment",
+                commands=[
+                    "copy requirements.txt /root/requirements.txt",
+                    "run python3 -m pip install -r requirements.txt",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    return commands
+
+
+def generate_dockerfile(pkg_root: Path, outfile: Path) -> None:
+    """
+    Generate a Dockerfile from files in the workflow directory.
+    """
+
+    print("Generating Dockerfile")
+    try:
+        with open(pkg_root / ".latch") as f:
+            config: LatchWorkflowConfig = LatchWorkflowConfig.from_json(f.read())
+            print("  - base image:", config.base_image)
+            print("  - latch version:", config.latch_version)
+    except FileNotFoundError:
+        raise RuntimeError(
+            "Could not find .latch file in supplied directory. If your workflow was created previously to release 2.13.0, you may need to run `latch init` to generate a .latch file."
```

Comment:
i'd rather have `.latch` be a directory which has the config file + generated ssh keys + other internal things

Reply:
I agree, that would be good.